python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -536,3 +563,67 @@ def to_arrow(self):
                 arrow_polygons,
             ],
         )
+
+    def contains(self, other, align=True):
+        if contains_only_polygons(self) is False:
+            raise TypeError("left series contains non-polygons.")
+
+        # RHS conditioning:
+        mode = "POINTS"
+        # point in polygon
+        if contains_only_linestrings(other) is True:
+            # condition for linestrings
+            mode = "LINESTRINGS"
+            xy = other.lines
+        elif contains_only_polygons(other) is True:
+            # polygon in polygon
+            mode = "POLYGONS"
+            xy = other.polygons
+        elif contains_only_multipoints(other) is True:
+            # mpoint in polygon
+            mode = "MULTIPOINTS"
+            xy = other.multipoints
+        else:
+            # no conditioning is required
+            xy = other.points
+            # mpoint in polygon
+            # linestring in polygon
```

Comment:
It is my intention to refactor this into an OO or possibly functional pattern as I write the next binop, as the abstraction will then be called for. I'd like to merge this as is. I'll create an issue from this comment.